best_practices/controllers.rst

```diff
@@ -164,13 +177,14 @@ manually. In our application, we have this situation in ``CommentController``:
 
 .. code-block:: php
 
+    use Doctrine\ORM\EntityManagerInterface;
+
     /**
      * @Route("/comment/{postSlug}/new", name = "comment_new")
      */
-    public function newAction(Request $request, $postSlug)
+    public function newAction(Request $request, $postSlug, EntityManagerInterface $em)
```

Comment:
I'm in favor to keep `$this->getDoctrine()->getManager()` it's a very common Controller feature that we need frequently, this looks to me like `$this->render()`, `$this->generateUrl()`, etc. I can't imagine having to inject these services too for Controllers.

Some situations:
* Using custom entity manager (`$this->getDoctrine()->getManager('custom')` or update the `services.yml` to inject the custom manager alias?)
* Getting entity manager from entity class (use `$this->getDoctrine()` for these cases only?)
* Getting entity repository (`$this->getDoctrine()->getRepository()` or Inject EM and then `$em->getRepository()`?).

I wondering If can I use `$this->getDoctrine()` for some situations why it's best type-hints the EM instead of `->Manager()`? To inject EM every time looks a bit overkill IMHO :)

Just I would recommend always use `$this->getDoctrine()->getManager()` instead of `$this->get('doctrine.orm.default_entity_manager')`.